tsdb/index/tsi1/index.go

```diff
@@ -707,7 +707,7 @@ func (i *Index) TagSets(name []byte, opt influxql.IteratorOptions) ([]*influxql.
 				}
 			}
 			// Associate the series and filter with the Tagset.
-			tagSet.AddFilter(string(SeriesElemKey(e)), e.Expr())
+			tagSet.AddFilter(string(models.MakeKey(e.Name(), e.Tags())), e.Expr())
```

Comment:
Yes, it's just for read operations. It needs to match the format that the query engine uses internally.